x/merkledb/trieview.go

```diff
@@ -567,11 +565,8 @@ func (t *trieView) commitChanges(ctx context.Context, trieToCommit *trieView) er
 
 // CommitToParent commits the changes from this view to its parent Trie
 func (t *trieView) CommitToParent(ctx context.Context) error {
-	// if we are about to write to the db, then we to hold the commitLock
-	if t.getParentTrie() == t.db {
-		t.db.commitLock.Lock()
-		defer t.db.commitLock.Unlock()
-	}
+	t.db.commitLock.Lock()
```

Comment:
Prevents multiple commits from occurring at the same time until I can figure out a way to manage the safety of this.